src/hooks/useEventCallback.ts

```diff
@@ -1,14 +1,12 @@
-import { useRef, useEffect, useCallback } from "react";
+import { useRef } from "react";
 
 // Saves incoming handler to the ref in order to avoid "useCallback hell"
-function useEventCallback<T>(handler?: (value: T) => void): (value: T) => void {
+export function useEventCallback<T>(handler?: (value: T) => void): (value: T) => void {
```

Comment:
When the function is called it already created a ref to the function. Which means one of the provided function will be GCed. 
So there is no difference for us which function we will call, so we don't even need the `if` clause here as well as `useEffect`. 
It makes this custom hook much faster

Reply:
Looks great!